ReSharper.FSharp/src/FSharp.Psi/src/Parsing/FSharpPreprocessedLexer.cs

```diff
@@ -0,0 +1,320 @@
+using System.Collections.Generic;
+using JetBrains.ReSharper.Plugins.FSharp.Psi.Parsing.Lexing;
+using JetBrains.ReSharper.Psi.ExtensionsAPI.Tree;
+using JetBrains.ReSharper.Psi.Parsing;
+using JetBrains.Text;
+using JetBrains.Util;
+
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Parsing
+{
+  public class FSharpPreprocessedLexer : ILexer<FSharpLexerState>
+  {
+    private TokenNodeType currTokenType;
+    private readonly FSharpLexer myLexer;
+    private readonly FSharpPreprocessor myPreprocessor;
+    private readonly HashSet<string> myDefinedConstants;
+    private PreprocessorState myState = new PreprocessorState();
```

Comment:
Lots of R# suggestions in file, like the one about making this field readonly. Were they ignored due to some reason?